### ISSUE TITLE
use SciPy to calculate ray tracking

### ISSUE THREAD

**Issue by yukimyg** [yukimyg/black-hole-image#7]
# Use Scipy to calculate ray tracking

## Summary
This issue is a proposal to verify that using a SciPy function instead of Numba's jit to compute ray tracing speeds up the process.

## Description
In this code, the process of calculating the ray trajectory numerically ([`runge_kutta` function in internal/calculation.py](https://github.com/yukimyg/black-hole-image/blob/main/internal/calculation.py#L12-L54)) is implemented natively and accelerated using [`jit` from `Numba`](https://numba.readthedocs.io/en/stable/user/parallel.html#numba-parallel). 
I suggest that we consider using [`scipy.integrate.solve_ivp`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html#scipy.integrate.solve_ivp) as an improvement.
In this case, we think the following points should be verified.
### Consideration
- Can we simply replace the current `runge_kutta` function with `scipy.integrate.solve_ivp`?
  - In other words, can `scipy.integrate.solve_ivp` run in a `jit` loop?
  - If not, how should we implement `scipy.integrate.solve_ivp`?
    - In other words, how to compensate for the fact that pixel-by-pixel calculations cannot be parallelized in `Numba` to keep up with performance.
- *Is it faster* than the original code?
